ui/src/api/index.ts

```diff
@@ -0,0 +1,16 @@
+import axios from "axios";
+import { API_URL } from "../config/constants";
+import { ICategory } from "../store/category";
+export interface Api {
+	categories(): Promise<ICategory>;
+}
+
+export class Hub implements Api {
+	async categories() {
+		try {
+			return axios.get(`${API_URL}/categories`);
+		} catch (err) {
+			return err.response;
```

Comment:
What happens when the API server is down or takes a long time to respond?

Reply:
Where would be the logic implemented for calling the API after some interval?

Reply:
So here we are handling and storing the error in the store whatever the error might and that error can you be used to display